dash/_jupyter.py

```diff
@@ -0,0 +1,496 @@
+import asyncio
+import io
+import inspect
+import logging
+import os
+import queue
+import uuid
+import re
+import sys
+import threading
+import time
+import traceback
+
+from typing_extensions import Literal
+
+from werkzeug.serving import make_server
+
+
+try:
+    from IPython import get_ipython
+    from IPython.display import IFrame, display, Javascript
+    from IPython.core.display import HTML
+    from IPython.core.ultratb import FormattedTB
+    from retrying import retry
+    from ansi2html import Ansi2HTMLConverter
+    import IPython
+    from ipykernel.comm import Comm
+    import nest_asyncio
+
+    import requests
+
+    _dash_comm = Comm(target_name="dash")
+    _dep_installed = True
+except ImportError:
+    _dep_installed = False
+    _dash_comm = None
+    get_ipython = None
```

Comment:
```suggestion
    get_ipython = lambda: None
```
Would simplify [usage](https://github.com/plotly/dash/pull/2530/files#diff-bf535dc399d2e2b82f75acf026efb0491402489814f432b433c3698bc61c2539R206) especially if we drop the bare [`import IPython`](https://github.com/plotly/dash/pull/2530/files#diff-bf535dc399d2e2b82f75acf026efb0491402489814f432b433c3698bc61c2539R26) (which I'm surprised the linter didn't complain about?) and reuse it in the [other](https://github.com/plotly/dash/pull/2530/files#diff-bf535dc399d2e2b82f75acf026efb0491402489814f432b433c3698bc61c2539R96) places we [call it](https://github.com/plotly/dash/pull/2530/files#diff-bf535dc399d2e2b82f75acf026efb0491402489814f432b433c3698bc61c2539R115)